CombinatorialGames/Surreal/Division.lean

```diff
@@ -291,4 +321,56 @@ theorem ratCast_strictMono : StrictMono ((↑) : ℚ → IGame) :=
 theorem ratCast_inj {m n : ℚ} : (m : IGame) = n ↔ m = n :=
   ratCast_strictMono.injective.eq_iff
 
+-- TODO: upstream
+attribute [simp] AntisymmRel.refl
+
+private theorem equiv_ratCast_of_mem_move_inv_natCast {n : ℕ} :
+    (∀ x ∈ leftMoves.{u} n⁻¹, ∃ q : ℚ, x ≈ q) ∧ (∀ x ∈ rightMoves.{u} n⁻¹, ∃ q : ℚ, x ≈ q) := by
+  cases n with
+  | zero => simp
+  | succ n =>
+    refine invRec (mod_cast n.succ_pos) ⟨0, ?_⟩ ?_ ?_ ?_ ?_
+    any_goals simp
+    all_goals
+      rintro _ hn rfl x hx q hq
+      use (1 + -q) / n
+      first | have := Numeric.of_mem_leftMoves hx | have := Numeric.of_mem_rightMoves hx
+      simp_all [invOption, ← Surreal.mk_eq_mk]
```

Comment:
```suggestion
private theorem equiv_ratCast_of_mem_move_inv_natCast {n : ℕ} :
    (∀ x ∈ leftMoves.{u} n⁻¹, ∃ q : ℚ, x ≈ q) ∧ (∀ x ∈ rightMoves.{u} n⁻¹, ∃ q : ℚ, x ≈ q) := by
  cases n with
  | zero => simp
  | succ n =>
    refine invRec (mod_cast n.succ_pos) ⟨0, ?_⟩ ?_ ?_ ?_ ?_
    any_goals (try · simp)
    all_goals
      simp_rw [Nat.cast_add, Nat.cast_one, leftMoves_natCast_succ, forall_exists_index]
      rintro _ hn rfl x hx q hq
      use (1 + -q) / n
      first | have := Numeric.of_mem_leftMoves hx | have := Numeric.of_mem_rightMoves hx
      simp_all [invOption, ← Surreal.mk_eq_mk]
```

Reply:
Wait, can't you just do `any_goals · simp`?

Reply:
That should work too!

Reply:
Doesn't work for whatever reason. I replaced this with `<;> (try · simp)`, since `any_goals` + `try` is redundant.